From eb6007f26dac9995a9a7fc0b88f1365013354e61 Mon Sep 17 00:00:00 2001
From: Florin Sarbu <florin@resin.io>
Date: Mon, 4 Jun 2018 12:36:24 +0200
Subject: [PATCH] artik533s machine specific integration of resin environment
 configuration

Upstream-Status: Inappropriate [configuration]

Signed-off-by: Florin Sarbu <florin@resin.io>
---
 include/configs/artik533_raptor.h | 30 +++++++++++++++++++-----------
 1 file changed, 19 insertions(+), 11 deletions(-)

diff --git a/include/configs/artik533_raptor.h b/include/configs/artik533_raptor.h
index 63d96c3..d8410d2 100644
--- a/include/configs/artik533_raptor.h
+++ b/include/configs/artik533_raptor.h
@@ -235,6 +235,7 @@
 #define CONFIG_CMD_GPT
 #define CONFIG_EFI_PARTITION
 #define CONFIG_PARTITION_UUIDS
+#define CONFIG_CMD_PART
 #define CONFIG_RANDOM_UUID
 
 #define CONFIG_CMD_DFU
@@ -390,7 +391,7 @@
 	"initrd_high=0xFFFFFFFF\0"	\
 	"load_fdt="							\
 		"if test $board_type = compy; then "			\
-			"ext4load mmc $rootdev:$bootpart $fdtaddr s5p4418-artik${model_id}-compy.dtb; "	\
+			"fatload mmc $rootdev:$bootpart $fdtaddr s5p4418-artik${model_id}-compy.dtb; "	\
 		"else "							\
 		"if test -z \"$fdtfile\"; then "                        \
 		"loop=$board_rev; "					\
@@ -401,15 +402,15 @@
 				"number=0$loop; "			\
 			"else number=$loop; "				\
 			"fi; "						\
-			"ext4size mmc $rootdev:$bootpart s5p4418-artik${model_id}-raptor-rev${number}.dtb && setexpr success 1; " \
+			"fatsize mmc $rootdev:$bootpart s5p4418-artik${model_id}-raptor-rev${number}.dtb && setexpr success 1; " \
 			"setexpr loop $loop - 1; "			\
 		"done; "					\
 		"if test $success -eq 0; then "				\
-			"ext4load mmc $rootdev:$bootpart $fdtaddr s5p4418-artik533-raptor-rev00.dtb;"	\
+			"fatload mmc $rootdev:$bootpart $fdtaddr s5p4418-artik533-raptor-rev00.dtb;"	\
 		"else "							\
-			"ext4load mmc $rootdev:$bootpart $fdtaddr s5p4418-artik${model_id}-raptor-rev${number}.dtb; "	\
+			"fatload mmc $rootdev:$bootpart $fdtaddr s5p4418-artik${model_id}-raptor-rev${number}.dtb; "	\
 		"fi; "							\
-		"else ext4load mmc $rootdev:$bootpart $fdtaddr $fdtfile; " \
+		"else fatload mmc $rootdev:$bootpart $fdtaddr $fdtfile; " \
 		"fi; setenv success; setenv number; setenv loop;"	\
 		"fi;\0"							\
 	"bootdelay=" __stringify(CONFIG_BOOTDELAY) "\0"			\
@@ -443,24 +444,24 @@
 		"factory_info write ethaddr $ethaddr;"			\
 		"run factory_save\0"					\
 	"load_args=run factory_load; setenv bootargs ${console} "	\
-		"root=/dev/mmcblk${rootdev}p${rootpart} ${root_rw} "	\
+		"${resin_kernel_root} ${root_rw} "			\
 		"rootfstype=${rootfs_type} ${recoverymode} ${ota} "	\
 		"drm_panel=$lcd_panel nr_cpus=${nr_cpus} ${opts} "	\
 		"bootfrom=${bootpart} rescue=${rescue} "		\
 		"board_type=${board_type};\0"				\
 	"load_kernel="							\
 		"ret=0; "						\
-		"ext4load mmc ${rootdev}:${bootpart} $kerneladdr $kernel_file && setexpr ret 1; " \
+		"fatload mmc ${rootdev}:${bootpart} $kerneladdr $kernel_file && setexpr ret 1; " \
 		"if test $ret -eq 0; then "				\
 			"if test $bootpart -eq 2; then "		\
 				"setenv bootpart 3; "			\
 			"else setenv bootpart 2; "			\
 			"fi; "						\
 			"setenv rescue 1; "				\
-			"ext4load mmc ${rootdev}:${bootpart} $kerneladdr $kernel_file; " \
+			"fatload mmc ${rootdev}:${bootpart} $kerneladdr $kernel_file; " \
 			"run load_args; "				\
 		"fi;\0"							\
-	"load_initrd=ext4load mmc ${rootdev}:${bootpart} $ramdiskaddr $ramdisk_file\0" \
+	"load_initrd=fatload mmc ${rootdev}:${bootpart} $ramdiskaddr $ramdisk_file\0" \
 	"boot_cmd_initrd="						\
 		"run load_kernel; run load_fdt; run load_initrd;"	\
 		"bootz $kerneladdr $ramdiskaddr $fdtaddr\0"		\
@@ -472,7 +473,14 @@
 		"setenv modulespart " __stringify(CONFIG_MODULES_PART_SD)"; "	\
 		"setenv rootpart " __stringify(CONFIG_ROOT_PART_SD)";\0"	\
 	"ramfsboot=run gen_addr; run load_args; run boot_cmd_initrd\0"	\
-	"mmcboot=run gen_addr; run load_args; run boot_cmd_mmcboot\0"	\
+	"mmcboot="							\
+		"run gen_addr;"						\
+		"setenv resin_kernel_load_addr ${kerneladdr};"		\
+		"run resin_set_kernel_root;"				\
+		"setenv rootdev ${resin_dev_index};"			\
+		"setenv rootpart ${resin_root_part};"			\
+		"setenv bootpart ${resin_boot_part};"			\
+		"run load_args; run boot_cmd_mmcboot\0"			\
 	"recovery_cmd=run sdrecovery; setenv recoverymode recovery\0"	\
 	"recoveryboot=run gen_sdrecaddr; run recovery_cmd; run ramfsboot\0"	\
 	"hwtestboot=setenv rootdev 1;"					\
@@ -480,6 +488,6 @@
 		"setenv fdtfile s5p4418-artik${model_id}-explorer.dtb;"		\
 		"run mmcboot\0"						\
 	"hwtest_recoveryboot=run recovery_cmd; run hwtestboot\0"        \
-	"bootcmd=run ramfsboot\0"
+	"bootcmd=run mmcboot\0"
 
 #endif /* __CONFIG_H__ */
-- 
2.7.4

